pkg/agent/config/configmgr.go

```diff
@@ -166,8 +168,9 @@ func (m *ConfigManager) Start(ctx context.Context) error {
 	if err = m.notifyListeners(); err != nil {
 		klog.ErrorS(err, "Failed to notify all listeners, retry by process chanagesQueue")
 	}
-
+	m.wg.Add(1)
```

Comment:
I think we'd better not change the main code. we can solve it in `FakeListener`.

Reply:
> I think we'd better not change the main code. we can solve it in `FakeListener`.

Yeah. but for this type of test, it is necessary to wait until all messages have been processed by the listeners before proceeding with the assertion. In this scenario, the listener lacks a mechanism to recognize when it will no longer be invoked.